crates/nu-engine/src/script.rs

```diff
@@ -117,44 +122,67 @@ pub fn process_script(
                     && Path::new(&name).is_dir()
                     && !ctx.host.lock().is_external_cmd(&name)
                 {
-                    // Here we work differently if we're in Windows because of the expected Windows behavior
-                    #[cfg(windows)]
-                    {
-                        if name.ends_with(':') {
-                            // This looks like a drive shortcut. We need to a) switch drives and b) go back to the previous directory we were viewing on that drive
-                            // But first, we need to save where we are now
-                            let current_path = ctx.shell_manager.path();
-
-                            let split_path: Vec<_> = current_path.split(':').collect();
-                            if split_path.len() > 1 {
-                                ctx.windows_drives_previous_cwd
-                                    .lock()
-                                    .insert(split_path[0].to_string(), current_path);
-                            }
+                    let tag = Tag {
+                        anchor: Some(AnchorLocation::Source(line.into())),
+                        span: name_span,
+                    };
+                    let path = {
```

Comment:
This code gained 1 level of nesting, therefore the diff shows more what actually happend. I commented lines that acutally changed.